r/NEWS.md

```diff
@@ -19,6 +19,65 @@
 
 # arrow 9.0.0.9000
 
+## Arrow dplyr queries
+
+Several new functions can be used in queries: 
+
+* `dplyr::across()` can be used to apply the same computation across multiple 
+  columns;
+* `add_filename()` can be used to get the filename a row came from (only 
+  available when querying `?Dataset`);
+* Added five functions in the `slice_*` family: `dplyr::slice_min()`, 
+  `dplyr::slice_max()`, `dplyr::slice_head()`, `dplyr::slice_tail()`, and
+  `dplyr::slice_sample()`.
+
+A full list of functions available in queries is available at `?acero`.
+
+A few new features and bugfixes were implemented for joins.
+Extension arrays are now supported in joins, allowing, for example, joining 
+datasets that contain [geoarrow](https://paleolimbot.github.io/geoarrow/) data.
+The `keep` argument is now supported, allowing separate columns for the left
+and right hand side join keys in join output. Full joins now coalesce the 
+join keys (when `keep = FALSE`), avoiding the issue where the join keys would 
+be all `NA` for rows in the right hand side without any matches on the left.
+
+A few breaking changes: Calling `dplyr::pull()` will return a `?ChunkedArray` 
+instead of an R vector. Calling `dplyr::compute()` on a query that is grouped 
+returns a `?Table`, instead of an query object.
+
+Finally, long-running queries can now be cancelled and will abort their 
+computation immediately.
+
+## Arrays and tables
+
+`as_arrow_array()` can now take `blob::blob` and `?vctrs::list_of`, which
+convert to binary and list arrays, respectively. Also fixed an issue where 
+`as_arrow_array()` ignored type argument when passed a `StructArray`.
+
+The `unique()` function works on `?Table`, `?RecordBatch`, `?Dataset`, and
+`?RecordBatchReader`.
+
+## Reading and writing
+
+`write_feather()` can take `FALSE` to choose writing uncompressed files.
+
+Also, a breaking change for IPC files in `write_dataset()`: passing
+`"ipc"` or  `"feather"` to `format` will now write files with `.arrow`
+extension instead of `.feather`.
```

Comment:
In the current version, if `"ipc"` is specified, the extension is `.ipc`, so perhaps the wording needs to be corrected?

Reply:
Are you sure about that?

https://github.com/apache/arrow/blob/63e1756778c8c7f0cfcd3cfca6774cd30ffc6c9d/r/R/dataset-write.R#L137-L139

https://github.com/apache/arrow/blob/63e1756778c8c7f0cfcd3cfca6774cd30ffc6c9d/r/tests/testthat/test-dataset-write.R#L169-L173

Reply:
```suggestion
extension instead of `.ipc` or `.feather`.
```